consoleApp/src/main/java/com/infoshare/view/NeedRequestView.java

```diff
@@ -23,4 +27,17 @@ public void showCreateNeedRequestMenu() {
         needRequestService.createNeedRequest(name, location, phone, typeOfHelp);
     }
 
+    public void pickupRequest (){
+        String inputCity = Util.readDataFromConsole("Z jakiego miasta chcesz się podjąć zgłoszenia?",
+                ValidatorEnum.POLISHSIGNS);
+        TypeOfHelp inputType = Util.createTypeOfHelp();
+        List<NeedRequest> filteredList = needRequestService.getNeedRequestFilteredList(inputCity,inputType);
+        if (filteredList.isEmpty()) {
+            System.out.println("Brak zgłoszeń pomocy o zadanych parametrach");
+        } else {
+            needRequestService.printNeedRequestsList(filteredList);
+            int choice = Util.readNumberFromUser("Którego zgłoszenia chcesz się podjąć?",filteredList.size()-1);
```

Comment:
tak, jak Magda napisała to nie działa. Mozemy uzywac tutaj UUID do identyfikacji needRequesta. Po przeniesieniu metody printNeedRequestList i jej poprawieniu kod może wyglądać tak:
```suggestion
           int choice = Util.readNumberFromUser("Którego zgłoszenia chcesz się podjąć?",filteredList.size());
            needRequestService.changeRequestStatus(filteredList.get( choice-1).getUuid());
        }
    }
    public void printNeedRequestsList(List<NeedRequest> needRequestList) {
        for (int i = 0; i < needRequestList.size(); i++) {
            System.out.println((i+1) + ". " + needRequestList.get(i).printDescription());
        }
    }
```

Reply:
jeszcze zmieniłam metodę printDescription w NeedRequest żeby zwracała Stringa